packages/loopring_v3/contracts/impl/libtransactions/WithdrawTransaction.sol

```diff
@@ -170,6 +170,23 @@ library WithdrawTransaction
             revert("INVALID_WITHDRAWAL_TYPE");
         }
 
+        // Check if there is a withdrawal recipient
+        address recipient = S.withdrawalRecipient[withdrawal.owner][withdrawal.to][withdrawal.tokenID][withdrawal.amount][withdrawal.nonce];
+        if (recipient != address(0)) {
+            // Auxiliary data is not supported
+            require (auxData.auxiliaryData.length == 0, "AUXILIARY_DATA_NOT_ALLOWED");
+
+            // Set the new recipient address
+            withdrawal.to = recipient;
+            // Allow any amount of gas to be used on this withdrawal (which allows the transfer to be skipped)
+            withdrawal.minGas = 0;
```

Comment:
Otherwise your idea won't work to do cheap redirects without actual token transfers. If the user does a normal withdrawal then surely in most cases some minimum amount of gas will be specified. So if we still keep enforcing this gas limit then the operator can't choose to use the cheaper path of just updating `amountWithdrawable` multiple times for the liquidity provider and he would still be forced to do the (likely) more expensive full token transfer withdrawals.